### PR TITLE
Remove useless static nameserver in sle15 autoyast template

### PR DESCRIPTION
**Pull request by Julie-CAO**
- Continue test even if logging script fail to be downloaded, take this case as example, https://openqa.suse.de/tests/12803055#step/sriov_network_card_pci_passthrough/147. Not sure the root of the dns problem, but test should continue even if no logs are uploaded. The failure could not be reproduced, it might be fixed by the other commit in this PR, or the dns resolution failure could occurred sporatically.
- Remove useless static nameserver in sle15. The nameserver was removed from sle12 host via https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18055

- Related ticket: https://progress.opensuse.org/issues/150848

Verification run:
https://openqa.suse.de/tests/12822812#step/sriov_network_card_pci_passthrough/149   //a simulated dns failure
https://openqa.suse.de/tests/12815611
https://openqa.suse.de/tests/12815613
https://openqa.suse.de/tests/12816981
https://openqa.suse.de/tests/12822714
https://openqa.suse.de/tests/12815601

### CODE REVIEW

**Comment by alice-suse** [comment]
> Continue test even if logging script fail to be downloaded, take this case as example, >https://openqa.suse.de/tests/12803055#step/sriov_network_card_pci_passthrough/147. Not sure the root of the dns problem, but test should continue even if no logs are uploaded. The failure could not be reproduced, it might be fixed by the other commit in this PR, or the dns resolution failure could occurred sporatically.

Thank you for the effort, Julie.
To me,  this PR is trying to make the test go further during network problems, which is good motivation. However, it brings the sacrifice that 
- when  in the end it still fails by sriov flaws, there may not be enough debug logs, then reviewer may still need to rerun the test after the review effort spent for this round
- more review challenges for others that do not know the details here: may confuse at the beginning, clear after checking out details

So, after several rounds of thinking,  I'd suggest below. 

The main principle is that when there is DNS problem or whatever network problem, it is fine to let the test fail at download_script.
- do not introduce the parameter to let download_execute_script function go when it fails, because it will make our test scripts loosing functions/logs
- but keep the debug info you added, eg that section of nslookup...
- if the issue is not reproducible, just let it go and make above merged.
- but if it is reproducible,  find the root cause and fix the problem that makes it download fail, but not let it hided/leaked

HDYT?

**Comment by Julie-CAO** [comment]
Whether the test is to continue or not, the log cannot be obtained because the logs fail to be uploaded to the worker server due to DNS issues. fg. in this case, https://openqa.suse.de/tests/12803055#downloads, there was no logs from sriov tests.

**Comment by Julie-CAO** [comment]
Whether the test is to continue or not, the log cannot be obtained because the logs fail to be uploaded to the worker server due to DNS issues. fg. in this case, https://openqa.suse.de/tests/12803055#downloads, there was no any logs(some logs were not affected by the download error) from sriov tests.

**Comment by alice-suse** [comment]
> Whether the test is to continue or not, the log cannot be obtained because the logs fail to be uploaded to the worker server due to DNS issues. fg. in this case, https://openqa.suse.de/tests/12803055#downloads, there was no any logs(some logs were not affected by the download error) from sriov tests.

I am afraid my points did not get to you well. I knew this when giving the review comment.

It is not important any more ;-). When thinking it again, I realize that maybe we are assuming the review process in very different ways when the test fail still in sriov after this PR workarounds the download part. It leads to very different pros and cons conclusions.

I was considering more that the review work for all possibilities of the job status would be more complex, especially after download fails, and sriov test fails again at the key part of plugged  in sriov nic connectivity test within vm ,because the network quality is hard to say and in addition to no debug log. Also the chance of this to happen should be much higher than the situation of problematic at download and recovered in the left sriov test period. So win < cost.

However, I realize that you maybe planning to review in very different way . If green let it go regardless of download failure. If red and there is download problem earlier, then no/limited analysis to the last failure point . The worst case would be to rerun, which is nearly the same as without this PR. Then this PR will mainly have pros. Right? If this is the case, I agree with the PR. But it is still valid that if the chance of such DNS problem increases, we need to find root cause and fix that. This is considered from the regard of decreasing rerun.

**Comment by Julie-CAO** [comment]
> However, I realize that you maybe planning to review in very different way . If green let it go regardless of download failure. If red and there is download problem earlier, then no/limited analysis to the last failure point . The worst case would be to rerun, which is nearly the same as without this PR. Then this PR will mainly have pros. Right? If this is the case, I agree with the PR. But it is still valid that if the chance of such DNS problem increases, we need to find root cause and fix that. This is considered from the regard of decreasing rerun.

Yes, exactly. the PR is to achieve both of the two targets: to make test continue and discover the dns failure.

 because test result is our ultimate goal, I'd like to get the final result even with some minor issues. The issue is not skipped in the test by the PR,  it is reported as a red info box and provide more helpful info.

**Comment by Julie-CAO** [comment]
Simulate a dns failure to validate the PR https://openqa.suse.de/tests/12822812#step/sriov_network_card_pci_passthrough/149

PR is ready.

**Review by alice-suse — APPROVED** [review]
LGTM

File: lib/virt_autotest/utils.pm [hunk 1]
```diff
@@ -362,17 +363,31 @@ sub download_script {
...
         record_info("URL is not accessible", "$script_url", result => 'fail') unless head($script_url);
         unless ($machine eq 'localhost') {
-            record_info("machine is not ssh accessible", "$machine", result => 'fail') unless script_run("ssh root\@$machine 'hostname'") == 0;
-            record_info("OSD is unaccessible from $machine", "that means the machine is having problem to access SUSE network", result => 'fail') unless script_run("ssh root\@$machine 'ping -c3 openqa.suse.de'") == 0;
+            # Have to output debug info at here because no logs will be uploaded if there are connection problems
+            if (script_run("ssh root\@$machine 'hostname'") == 0) {
+                script_run("ssh root\@$machine 'ping -c3 openqa.suse.de'");
+                script_run("ssh root\@$machine 'nslookup " . get_var('WORKER_HOSTNAME', 'openqa.suse.de') . "'");
+                script_run("ssh root\@$machine 'cat /etc/resolv.conf'");
+            }
```
Comment (waynechen55) [inline lib/virt_autotest/utils.pm:379]: Maybe you also need ```dnsdomainname``` if take the nature of the problem into consideration.

Comment (waynechen55) [reply lib/virt_autotest/utils.pm:379]: For your download logic here, I would suggest using curl to check url availability in the first place before starting downloading. ```curl -I``` will also returns http status code.

Comment (Julie-CAO) [reply lib/virt_autotest/utils.pm:382]: Even if the URL does not exist, curl can always get some content. fg. https://openqa.suse.de/tests/12816981#step/sriov_network_card_pci_passthrough/140

Comment (Julie-CAO) [reply lib/virt_autotest/utils.pm:382]: > Maybe you also need `dnsdomainname` if take the nature of the problem into consideration.

added. thanks.

Comment (waynechen55) [reply lib/virt_autotest/utils.pm:382]: You need to use ```curl -I```, for example:
```
waynechen:~ # curl -I http://does-not-exist.com
HTTP/1.1 302 Found
date: Fri, 17 Nov 2023 07:06:16 GMT
server: Apache
set-cookie: __tad=1700204776.8363663; expires=Mon, 14-Nov-2033 07:06:16 GMT; Max-Age=315360000
location: http://ww25.does-not-exist.com/?subid1=20231117-1806-1613-9bef-97bd32ee250b
content-type: text/html; charset=UTF-8
connection: close
```

Comment (Julie-CAO) [reply lib/virt_autotest/utils.pm:382]: > For your download logic here, I would suggest using curl to check url availability in the first place before starting downloading. `curl -I` will also returns http status code.

The URL is checked on host via
`record_info("URL is not accessible", "$script_url", result => 'fail') unless head($script_url);`.

Do you mean move this line ahead before run curl inside the guest?
`unless (script_retry($cmd, timeout => 900, retry => 2, die => 0) == 0) {`

Comment (Julie-CAO) [reply lib/virt_autotest/utils.pm:390]: > > Maybe you also need `dnsdomainname` if take the nature of the problem into consideration.
> 
> added. thanks.

dnsdomainname: `name or service not known` in guest, 
https://openqa.suse.de/tests/12849090#step/sriov_network_card_pci_passthrough/148

Comment (waynechen55) [reply lib/virt_autotest/utils.pm:390]: OK.

Comment (Julie-CAO) [reply lib/virt_autotest/utils.pm:390]: > > For your download logic here, I would suggest using curl to check url availability in the first place before starting downloading. `curl -I` will also returns http status code.
> 
> The URL is checked on host via `record_info("URL is not accessible", "$script_url", result => 'fail') unless head($script_url);`.
> 
> Do you mean move this line ahead before run curl inside the guest? `unless (script_retry($cmd, timeout => 900, retry => 2, die => 0) == 0) {`

Move the URL checking ahead, and verification run passed. http://openqa.suse.de/tests/12849138#step/sriov_network_card_pci_passthrough/138

**Comment by Julie-CAO** [comment]
Removed the debugging codes and the PR is ready.

**Pull request merged**